ui/cryptomaterial/segmented_control.go

```diff
@@ -0,0 +1,107 @@
+package cryptomaterial
+
+import (
+	"gioui.org/font"
+	"gioui.org/layout"
+
+	"github.com/crypto-power/cryptopower/ui/values"
+)
+
+type SegmentedControl struct {
+	theme *Theme
+	list  *ClickableList
+
+	selectedIndex int
+	segmentTitles []string
+
+	changed bool
+}
+
+func (t *Theme) SegmentedControl(segmentTitles []string) *SegmentedControl {
+	list := t.NewClickableList(layout.Horizontal)
+	list.IsHoverable = false
+
+	return &SegmentedControl{
+		list:          list,
+		theme:         t,
+		segmentTitles: segmentTitles,
+	}
+}
+
+func (sc *SegmentedControl) Layout(gtx C) D {
+	sc.handleEvents()
+
+	return LinearLayout{
+		Width:      WrapContent,
+		Height:     WrapContent,
+		Background: sc.theme.Color.Background,
+		Border:     Border{Radius: Radius(8)},
+	}.Layout(gtx,
+		layout.Rigid(func(gtx C) D {
+			return sc.list.Layout(gtx, len(sc.segmentTitles), func(gtx C, i int) D {
+				isSelectedSegment := sc.SelectedIndex() == i
+				return layout.Stack{Alignment: layout.Center}.Layout(gtx,
+					layout.Stacked(func(gtx C) D {
+						return layout.Inset{}.Layout(gtx, func(gtx C) D {
+							return layout.Center.Layout(gtx, func(gtx C) D {
+								bg := sc.theme.Color.SurfaceHighlight
+								txt := sc.theme.Label(values.TextSize16, sc.segmentTitles[i])
+								txt.Color = sc.theme.Color.GrayText1
+								txt.Font.Weight = font.SemiBold
+								border := Border{Radius: Radius(0)}
+								if isSelectedSegment {
+									bg = sc.theme.Color.Surface
+									txt.Color = sc.theme.Color.Text
+									border = Border{Radius: Radius(8)}
+								}
+								return LinearLayout{
+									Width:      WrapContent,
+									Height:     WrapContent,
+									Padding:    layout.UniformInset(values.MarginPadding8),
+									Background: bg,
+									Margin:     layout.UniformInset(values.MarginPadding5),
+									Border:     border,
+								}.Layout(gtx,
+									layout.Rigid(func(gtx C) D {
+										return txt.Layout(gtx)
+									}),
+								)
```

Comment:
This is basically a text with padding, margin, background and a border. Can we have a custom widget for this? We can probably call it `Label` or `DecoratedText` or something; it can take a background color, padding, margin and border. Even text color and font as well.